include/envoy/ssl/context_config.h

```diff
@@ -16,6 +17,40 @@
 namespace Envoy {
 namespace Ssl {
 
+class TlsCertificateConfigProvidersFactory {
+public:
+  virtual ~TlsCertificateConfigProvidersFactory() = default;
+
+  /**
+   * Create the certificate config providers factory
+   */
+  virtual std::vector<Secret::TlsCertificateConfigProviderSharedPtr> create() PURE;
+};
+
+using TlsCertificateConfigProvidersFactoryPtr =
+    std::unique_ptr<TlsCertificateConfigProvidersFactory>;
+
+class CertificateValidationContextConfigProviderFactory {
+public:
+  virtual ~CertificateValidationContextConfigProviderFactory() = default;
+
+  /**
+   * Create the validation context config provider.
+   */
+  virtual Secret::CertificateValidationContextConfigProviderSharedPtr create() PURE;
+
+  /**
+   * To get default certificate validation context. This value is nullptr by default except for
```

Comment:
wording/grammar: "Get the default certificate validation context. This value is nullptr by default, except for when the validation context type is a combined validation context." Or similar.